source/Octopus.Tentacle.Contracts/ClientServices/IClientCapabilitiesServiceV2.cs

```diff
@@ -0,0 +1,10 @@
+using Halibut.ServiceModel;
+using Octopus.Tentacle.Contracts.Capabilities;
+
+namespace Octopus.Tentacle.Client.ClientServices
+{
+    public interface IClientCapabilitiesServiceV2
+    {
+        public CapabilitiesResponseV2 GetCapabilities(HalibutProxyRequestOptions halibutProxyRequestOptions);
```

Comment:
I'm struggling to wrap my head around Client services. It's unclear to me what problem they are trying to solve and how they differ from the services they map to. 

Also, it is not clear to me why we have introduced `halibutProxyRequestOptions`  on the client services instead of adding a cancellation token to the V2 interface and how would they work with cancellation token that we will introduce as part of Async Halibut work.

Reply:
This Halibut [PR](https://github.com/OctopusDeploy/Halibut/pull/238) may help answer some of those questions.

Reply:
A change has recently been made to Halibut to try and improve connecting cancellation tokens. This is a cancellation token passed to a Halibut Client that is honoured when a client is trying to connect to a listening tentacle or a polling request has been queued but not yet dequeued by tentacle. After those events, and when the request is "in-flight" the connecting cancellation token is no longer used.

The issue we previously had was that this connecting cancellation token was provided to the creator of the Client so a single cancellation token would control every RPC call made using that client. Once the connecting cancellation token was cancelled, the Clients was unusable for new requests (as they would immediately cancel).

We had an implementation using this Client wide Cancellation Token but it was messy, we had to have a script service v2 with a Cancellation token and then a second script service v2 without a cancellation token so we could continue to make RPC calls even after the Execution of the Script had been cancelled and the connecting cancellation token cancelled.

The current TentacleRemoteEndpointFacade and ScriptObserver does not suffer from this issue because it is not very good at cancelling requests when the execution is cancelled. We have improved on this and hit this issue. 

Client Services were introduced so that the interface for the client and the interface for the Service could diverge and allow additional parameters to be passed to the client that doesn't go over the wire to the service.  Example the connecting cancellation token. We now allow it to be passed for each RPC call but this can't/shouldn't go over the wire to Tentacle, it's client side only.

HalibutProxyRequestOptions is a container for the client specific parameters e.g. connect cancellation token.